src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetRandomCommand.cs

```diff
@@ -396,11 +396,14 @@ protected override void BeginProcessing()
                         this.ThrowMinGreaterThanOrEqualMax(min, max);
                     }
 
-                    int randomNumber = this.Generator.Next(min, max);
-                    Debug.Assert(min <= randomNumber, "lower bound <= random number");
-                    Debug.Assert(randomNumber < max, "random number < upper bound");
+                    for (int i = 0; i < Math.Max(1, this.Count); i++)
+                    {
+                        int randomNumber = this.Generator.Next(min, max);
+                        Debug.Assert(min <= randomNumber, "lower bound <= random number");
```

Comment:
I don't think we generally use `Debug.Assert()` in cmdlets. This seems like something that should only fail if the `Random` class has a fatal regression for its `Next()` code path(s), which is outside the purview of our debug code in general, I would think.

I think if we want to test this at all, we should implement this as a Pester test so that we can see it in CI rather than waiting for someone to stumble across it. 🙂

Reply:
Hmm not too sure I agree that Debug.Assert isn't used since I have found it in quite a few cmdlets throughout the codebase.
I do agree that ideally this type of thing should surface as a result of a failing CI test but I don't know how important the consequence of removing it would be (scripts not crashing and using potentially invalid random values). 🙂

@JamesWTruher  ? @PaulHigin ?

Reply:
If I remember rightly @PaulHigin,  aren't Debug.Assert() calls ignored / no-op in `release` builds anyway? 🤔 Or am I misremembering? 😄

Reply:
I agree that this is the wrong place for debug asserts and the correct place is in the random number generator adapter.  Debug asserts are only evaluated in debug builds and are intended to verify expected (and assumed) behavior.

Reply:
Done :)